File: thread_locks/rust/hello_thread.rs
// Simple 'hello' thread example.
// Mainly inspired from official book at
// http://doc.rust-lang.org/book/concurrency.html

use std::thread;

fn main() {
    println!("Hello from main");
    thread::spawn(|| {
        println!("Hello from new thread");
    });
}

File: thread_locks/python/hello_thread.py
# -*- coding: utf-8 -*-

"""Simple 'hello' thread example.
"""

from __future__ import print_function
import threading


def hello():
    print("Hello from new thread")

def main():
    print("Hello from main")
    threading.Thread(target=hello).start()

if __name__ == '__main__':
    main()
